shims.c

```diff
@@ -0,0 +1,20 @@
+#include <assert.h>
+#include <stdint.h>
+
+#include "shims.h"
+
+#if defined(_WIN32)
+
+#include "safewindows.h"
+
+int getpagesize(void) {
```

Comment:
This API is also not standard, so we could just put this with the platform-specific code at the top of the one file that needs it.